plugins/samples/html_domain_rewrite/tests.textpb

```diff
@@ -0,0 +1,121 @@
+                    "<p>This is a paragraph.</p>"
+                    "<div>"
+                    "<a href=\"https://bar.com\">website</a>"
+                    "</div>"
+                    "<a href=\"https://bar.com\">website</a>"
+                    "</body>"
+                    "</html>"
+      }
+    }
+  }
+}
+
+test {
+  name: "DomainRewriteMultipleChunks"
+  num_chunks:10
+  response_body {
+    input { content:  "<!DOCTYPE html>"
+                      "<html>"
+                      "<head>"
+                      "<title>Page Title</title>"
+                      "</head>"
+                      "<body>"
+
+                      "<h1>This is a Heading</h1>"
+                      "<p>This is a paragraph.</p>"
+                      "<div>"
+                      "<a href=https://foo.com>website</a>"
+                      "</div>"
+                      "<a href=https://foo.com>website</a>"
+                      "</body>"
+                      "</html>"
+    }
+    result {
+      body { exact: "<!DOCTYPE html>"
+                    "<html>"
+                    "<head>"
+                    "<title>Page Title</title>"
+                    "</head>"
+                    "<body>"
+
+                    "<h1>This is a Heading</h1>"
+                    "<p>This is a paragraph.</p>"
+                    "<div>"
+                    "<a href=\"https://bar.com\">website</a>"
+                    "</div>"
+                    "<a href=\"https://bar.com\">website</a>"
+                    "</body>"
+                    "</html>"
+      }
+    }
+  }
+}
+
+test {
+  name: "DomainRewriteFooOutside<a>"
+  num_chunks:10
+  response_body {
+    input { content:  "<!DOCTYPE html>"
+                      "<html>"
+                      "<head>"
+                      "<title>Page Title</title>"
+                      "</head>"
+                      "<body>"
+
+                      "<h1>This is a Heading</h1>"
+                      "<p>This is a paragraph. Instances of \"foo\" here should not change</p>"
+                      "<div id=foo>"
+                      "<a href=https://foo.com>website</a>"
+                      "</div>"
+                      "<a href=https://foo.com>website</a>"
+                      "</body>"
+                      "</html>"
+    }
+    result {
+      body { exact: "<!DOCTYPE html>"
+                    "<html>"
+                    "<head>"
+                    "<title>Page Title</title>"
+                    "</head>"
+                    "<body>"
+
+                    "<h1>This is a Heading</h1>"
+                    "<p>This is a paragraph. Instances of \"foo\" here should not change</p>"
+                    "<div id=foo>"
+                    "<a href=\"https://bar.com\">website</a>"
+                    "</div>"
+                    "<a href=\"https://bar.com\">website</a>"
+                    "</body>"
+                    "</html>"
+      }
+    }
+  }
+}
```

Comment:
I'd probably just merge these 3 tests together for readability / clarity. The chunking is something you always want to test, and the "should not change" comment is nicely self explanatory.